packages/contractkit/src/identity/offchain/utils.ts

```diff
@@ -0,0 +1,350 @@
+import { Err, Ok, parseJsonAsResult, Result, trimLeading0x } from '@celo/base/src'
+import { Address, publicKeyToAddress } from '@celo/utils/lib/address'
+import { AES128Decrypt, AES128Encrypt, Encrypt, IV_LENGTH } from '@celo/utils/lib/ecies'
+import { EIP712Object, EIP712TypedData } from '@celo/utils/lib/sign-typed-data-utils'
+import { ensureUncompressed, trimUncompressedPrefix } from '@celo/utils/src/ecdh'
+import { createHmac, randomBytes } from 'crypto'
+import { keccak256 } from 'ethereumjs-util'
+import { isLeft } from 'fp-ts/lib/Either'
+import * as t from 'io-ts'
+import { join } from 'path'
+import OffchainDataWrapper, { OffchainErrors, OffchainErrorTypes } from '../offchain-data-wrapper'
+import {
+  InvalidDataError,
+  InvalidKey,
+  OffchainError,
+  SchemaErrors,
+  SchemaErrorTypes,
+  UnavailableKey,
+} from './accessors/errors'
+
+const KEY_LENGTH = 16
+
+// label = PRF(ECDH(A, B), A || B || data path)
+// ciphertext path = "/cosmetic path/" || base64(label)
+function getCiphertextLabel(
+  path: string,
+  sharedSecret: Buffer,
+  senderPublicKey: string,
+  receiverPublicKey: string
+) {
+  const senderPublicKeyBuffer = Buffer.from(trimLeading0x(senderPublicKey), 'hex')
+  const receiverPublicKeyBuffer = Buffer.from(trimLeading0x(receiverPublicKey), 'hex')
```

Comment:
Actually one more thing. We need to ensure these are in a consistent encoding, either compressed or uncompressed (If uncompressed we can update the spec) because the on-chain Account does not specify.

Reply:
ICYMI https://github.com/celo-org/celo-monorepo/pull/5091#discussion_r509317862. We support either. Do we want to force one or the other?

Reply:
Yes. My point here is that IIUC, [the caller](https://github.com/celo-org/celo-monorepo/pull/5091/files/3144e537d9279ea17ce4e73daadd7fe90fa1a7c1#diff-e7d55cae43eddd87cf3d520332a28a94c3259940b5d1f8e77355eb1a73d8ab1eR66) of this function is passing in the direct output of the view call to retrieve the DEK. There needs to be a normalization step at some point between retrieving it from the chain and [concatenating it with the shared secret](https://github.com/celo-org/celo-monorepo/pull/5091/files/3144e537d9279ea17ce4e73daadd7fe90fa1a7c1#diff-e7d55cae43eddd87cf3d520332a28a94c3259940b5d1f8e77355eb1a73d8ab1eR35) as input to the MAC.

Reply:
Ah I see what you're talking about now. Good point. I've used ensureCompressed as input to the MAC